tonic/src/metadata/value.rs

```diff
@@ -282,6 +273,161 @@ impl<VE: ValueEncoding> MetadataValue<VE> {
     }
 }
 
+/// Attempt to convert a byte slice to a `MetadataValue`.
+///
+/// For Ascii metadata values, If the argument contains invalid metadata
+/// value bytes, an error is returned. Only byte values between 32 and 255
+/// (inclusive) are permitted, excluding byte 127 (DEL).
+///
+/// For Binary metadata values this method cannot fail. See also the Binary
+/// only version of this method `from_bytes`.
+///
+/// # Examples
+///
+/// ```
+/// # use tonic::metadata::*;
+/// let val = AsciiMetadataValue::try_from_bytes(b"hello\xfa").unwrap();
+/// assert_eq!(val, &b"hello\xfa"[..]);
+/// ```
+///
+/// An invalid value
+///
+/// ```
+/// # use tonic::metadata::*;
+/// let val = AsciiMetadataValue::try_from_bytes(b"\n");
+/// assert!(val.is_err());
+/// ```
+impl<'a, VE: ValueEncoding> TryFrom<&'a [u8]> for MetadataValue<VE> {
```

Comment:
Can this lifetime be elided?

Reply:
It can be, but then Rustdoc generates docs with a signature like `TryFrom<&'0 str>`. If I name the lifetime, then it outputs the signature `TryFrom<&'a str>` which I think is more readable, more what Rust programmers are used to seeing.

I copied this style of doing it [from the `hyper` docs](https://docs.rs/hyper/latest/hyper/header/struct.HeaderValue.html#impl-TryFrom%3C%26%27a%20str%3E) -- happy to change it to the elided lifetime if you prefer, but I thought matching Hyper's docs is a good tiebreaker when I wasn't sure which was better.

Reply:
Oh cool didn't know that, seems fine to me!